Mathlib/RingTheory/JacobsonIdeal.lean

```diff
@@ -48,6 +54,175 @@ Jacobson, Jacobson radical, Local Ideal
+
+variable {f}
+
+theorem map_jacobson_of_ker_le (surj : Function.Surjective f)
+    (le : LinearMap.ker f ≤ jacobson R M) :
+    map f (jacobson R M) = jacobson R₂ M₂ :=
+  le_antisymm (map_jacobson_le f) <| by
+    rw [jacobson, sInf_eq_iInf'] at le
+    conv_rhs => rw [jacobson, sInf_eq_iInf', map_iInf_of_ker_le surj le]
+    exact le_iInf fun m ↦ sInf_le (isCoatom_map_of_ker_le surj (le_iInf_iff.mp le m) m.2)
+
+theorem comap_jacobson_of_ker_le (surj : Function.Surjective f)
+    (le : LinearMap.ker f ≤ jacobson R M) :
+    comap f (jacobson R₂ M₂) = jacobson R M := by
+  rw [← map_jacobson_of_ker_le surj le, comap_map_eq_self le]
+
+theorem map_jacobson_of_bijective (hf : Function.Bijective f) :
+    map f (jacobson R M) = jacobson R₂ M₂ :=
+  map_jacobson_of_ker_le hf.2 <| by simp_rw [(LinearMapClass.ker_eq_bot _).mpr hf.1, bot_le]
+
+theorem comap_jacobson_of_bijective (hf : Function.Bijective f) :
+    comap f (jacobson R₂ M₂) = jacobson R M :=
+  comap_jacobson_of_ker_le hf.2 <| by simp_rw [(LinearMapClass.ker_eq_bot _).mpr hf.1, bot_le]
+
+theorem jacobson_quotient_of_le {N : Submodule R M} (le : N ≤ jacobson R M) :
+    jacobson R (M ⧸ N) = map N.mkQ (jacobson R M) :=
+  (map_jacobson_of_ker_le N.mkQ_surjective <| by rwa [ker_mkQ]).symm
+
+theorem jacobson_le_of_eq_bot {N : Submodule R M} (h : jacobson R (M ⧸ N) = ⊥) :
+    jacobson R M ≤ N := by
+  simp_rw [← N.ker_mkQ, ← comap_bot, ← h, le_comap_jacobson]
+
+variable (R M)
+
+theorem jacobson_quotient_jacobson : jacobson R (M ⧸ jacobson R M) = ⊥ := by
+  rw [jacobson_quotient_of_le le_rfl, mkQ_map_self]
+
+theorem jacobson_lt_top [Nontrivial M] [IsCoatomic (Submodule R M)] : jacobson R M < ⊤ := by
+  obtain ⟨m, hm, -⟩ := (eq_top_or_exists_le_coatom (⊥ : Submodule R M)).resolve_left bot_ne_top
+  exact (sInf_le <| Set.mem_setOf.mpr hm).trans_lt hm.1.lt_top
+
+example [Nontrivial M] [Module.Finite R M] : jacobson R M < ⊤ := jacobson_lt_top R M
+
+variable {ι} (M : ι → Type*) [∀ i, AddCommGroup (M i)] [∀ i, Module R (M i)]
+
+theorem jacobson_pi_le : jacobson R (Π i, M i) ≤ Submodule.pi Set.univ (jacobson R <| M ·) := by
+  simp_rw [← iInf_comap_proj, jacobson, sInf_eq_iInf', comap_iInf, le_iInf_iff]
+  intro i m
+  exact iInf_le_of_le ⟨_, (isCoatom_comap_iff <| LinearMap.proj_surjective i).mpr m.2⟩ le_rfl
+
+/-- A product of modules with trivial Jacobson radical (e.g. simple modules) also has trivial
+Jacobson radical. -/
+theorem jacobson_pi_eq_bot (h : ∀ i, jacobson R (M i) = ⊥) : jacobson R (∀ i, M i) = ⊥ :=
+  le_bot_iff.mp <| (jacobson_pi_le R M).trans <| by simp_rw [h, pi_univ_bot, le_rfl]
+
+end Module
+
+section
+
+variable (R R₂ : Type*) [Ring R] [Ring R₂] (f : R →+* R₂) [RingHomSurjective f]
+variable (M : Type*) [AddCommGroup M] [Module R M]
+
+namespace Ring
+
+abbrev jacobson : Ideal R := Module.jacobson R R
+
+instance : (jacobson R).IsTwoSided :=
+  ⟨fun b ha ↦ Module.le_comap_jacobson (f := LinearMap.toSpanSingleton R R b) ha⟩
```

Comment:
@Vtec234 Here's an easy proof that the Jacobson radical of a ring is a two-sided ideal, using Jacobson radicals of modules.

Reply:
Neat! Can it be used to simplify the proof of `jacobson_mul_mem_right` below?